File: backend/src/main/java/com/questbase/backend/entity/Quest.java
package com.questbase.backend.entity;

import java.time.LocalDateTime;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "quests")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Quest {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 150)
    private String title;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column(nullable = false)
    private String status;

    private String difficulty;

    @Column(name = "reward_xp")
    private Integer rewardXp;

    @Column(name = "created_at", insertable = false, updatable = false)
    private LocalDateTime createdAt;
}

File: frontend/src/types/api/quest.ts
import { Campaign } from "./campaign";

export interface Quest {
  id: number,
  title: string,
  description: string,
  status: string,
  difficulty: string,
  rewardXp: string,
  createdAt: string,
  campaign: Campaign
}

export interface CreateQuestRequest {
  title: string,
  description: string,
  status: string,
  difficulty: string,
  rewardXp: number,
  campaignId: number
}